File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/api/SectionApi.java
package ru.spring5.elena.librarymanager.api;

import java.util.List;

import javax.validation.constraints.NotNull;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ru.spring5.elena.librarymanager.api.dto.Edition;
import ru.spring5.elena.librarymanager.api.dto.Section;

public interface SectionApi {

	Section getSection(
			@NotNull Long sectionId
	);

	Page<Section> searchRootSections(
			Pageable pageable
	);

	Page<Section> searchSectionsByParentSectionId(
			@NotNull Long sectionId, 
			Pageable pageable
	);

	Page<Edition> searchEditionsBySectionId(
			@NotNull Long sectionId, 
			Pageable pageable
	);

	List<Edition> updateEditionsBySectionId(
			@NotNull Long sectionId, 
			@NotNull List<Long> editionIds
	);

}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/SectionApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.Edition
import ru.elenacherev.librarymanager.api.dto.Section
import java.util.*

interface SectionApi {
    fun getSection(
        sectionId: UUID
    ): Section?

    fun searchRootSections(
        pageable: Pageable
    ): Page<Section>

    fun searchSectionsByParentSectionId(
        sectionId: UUID,
        pageable: Pageable
    ): Page<Section>

    fun searchEditionsBySectionId(
        sectionId: UUID,
        pageable: Pageable
    ): Page<Edition>

    fun updateEditionsBySectionId(
        sectionId: UUID,
        editionIds: List<UUID>
    ): List<Edition>
}